cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -240,6 +243,15 @@ StrptimeOptions::StrptimeOptions(std::string format, TimeUnit::type unit)
 StrptimeOptions::StrptimeOptions() : StrptimeOptions("", TimeUnit::SECOND) {}
 constexpr char StrptimeOptions::kTypeName[];
 
+StrftimeOptions::StrftimeOptions(std::string format, std::string timezone)
+    : FunctionOptions(internal::kStrftimeOptionsType),
+      format(std::move(format)),
+      timezone(std::move(timezone)) {
+  tz = arrow_vendored::date::locate_zone(this->timezone);
+}
+StrftimeOptions::StrftimeOptions() : StrftimeOptions("%Y-%m-%dT%H:%M:%S", "UTC") {}
```

Comment:
Should the default format string include the trailing `%z` so that it is ISO-8601 compliant?

Reply:
Indeed. Will add.

Reply:
I don't think that `%z` adds a trailing z as in the ISO format. It adds a numeric offset (eg https://www.cplusplus.com/reference/ctime/strftime/):

```
In [44]: datetime.datetime(2012, 1, 1, tzinfo=datetime.timezone.utc).strftime("%Y-%m-%dT%H:%M:%S%z")
Out[44]: '2012-01-01T00:00:00+0000'
```

Reply:
According to [wikipedia](https://en.wikipedia.org/wiki/ISO_8601) these are valid formats: 
1. 2021-07-15T11:24:54+00:00
1. 2021-07-15T11:24:54Z
1. 20210715T112454Z

`%z` will give us option 1. I don't have a strong preference and pandas doesn't seem to have a default.

Reply:
If it is significant effort I agree that `+00:00` is good enough but I personally prefer `Z` if there is a quick option to enable it when the time zone is UTC.

Reply:
Switched to `%Y-%m-%dT%H:%M:%SZ` (2.) as default.